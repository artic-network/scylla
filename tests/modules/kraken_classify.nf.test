nextflow_workflow {

    name "Kraken classify module tests"
    script "modules/kraken_classification.nf"
    workflow "kraken_classify"
    tag "unit"
    tag "ci"

    test("kraken2_classify_success") {

        when {
            params {
                kraken_database = [
                    "default": [
                        "name": "testdb-kraken2",
                        "path": "${projectDir}/tests/test-data/database/testdb-kraken2",
                        "host": "localhost",
                        "port": 8080,
                    ]
                ]
            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "test",
                        file("${projectDir}/tests/test-data/barcode01/barcode01.fq.gz", checkIfExists: true)
                    )
                )
                input[1] = "default"
                input[2] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }
}