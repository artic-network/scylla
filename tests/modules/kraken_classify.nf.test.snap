{
    "kraken2_classify_success": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "testdb-kraken2",
                        "testdb-kraken2.kraken_assignments.tsv:md5,4e0ff0c093a962377307c7a547c852c6"
                    ]
                ],
                "1": [
                    [
                        "test",
                        "testdb-kraken2",
                        "testdb-kraken2.kraken_report.txt:md5,589379bb70d49b40be3973eb42f03b24"
                    ]
                ],
                "assignments": [
                    [
                        "test",
                        "testdb-kraken2",
                        "testdb-kraken2.kraken_assignments.tsv:md5,4e0ff0c093a962377307c7a547c852c6"
                    ]
                ],
                "kreport": [
                    [
                        "test",
                        "testdb-kraken2",
                        "testdb-kraken2.kraken_report.txt:md5,589379bb70d49b40be3973eb42f03b24"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-16T08:14:04.781506718"
    }
}